.github/workflows/tests.yml

```diff
@@ -0,0 +1,77 @@
+name: vHive tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+env:
+  GOOS: linux
+  GO111MODULE: on
+
+jobs:
+  test:
+    name: Integration tests on Github Runners
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        module: [taps, misc]
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: go build -race -v -a ./...
+
+    - name: Run tests in submodules
+      env:
+          MODULE: ${{ matrix.module }}
+      run: |
+        make -C $MODULE test
+        make -C $MODULE test-man
+  
+  ctriface-test:
+    name: ctriface & end-to-end integration tests on self hosted runners
+    runs-on: self-hosted
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: go build -race -v -a ./...
+
+    - name: Run tests in submodules
+      run: |
+        make -C ctriface test
+        make -C ctriface test-man
+
+    - name: Cleaning
+      if: ${{ always() }}
+      run: ./scripts/clean_fcctr.sh
+
+    - name: Run vHive end-to-end tests
+      run: make test
+
+    - name: Cleaning
+      if: ${{ always() }}
+      run: ./scripts/clean_fcctr.sh
+
+    - name: Run vHive tests that need manual cleaning
```

Comment:
rename: `vHive Integration tests (with manual cleaning)`. Split this into a separate job

Reply:
Done. Separated integrations tests from unit tests